db/permission/permission.go

```diff
@@ -0,0 +1,35 @@
+// Copyright 2024 Democratized Data Foundation
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+/*
+Package acp utilizes the sourcehub acp module to bring the functionality to defradb, this package also helps
+avoid the leakage of direct sourcehub references through out the code base, and eases in swapping
+between local embedded use case and a more global on sourcehub use case.
+*/
+
+package permission
+
+import (
+	"github.com/sourcenetwork/defradb/client"
+)
+
+// IsPermissioned returns true if the collection has a policy, otherwise returns false.
+//
+// This tells us if access control is enabled for this collection or not.
+func IsPermissioned(collection client.Collection) (string, string, bool) {
```

Comment:
todo: This really feels like a waste of the `Option` type, this whole function should be replaced with `policy.HasValue()` IMO

Reply:
I have mixed feelings on this one, I like the utility of it rather than a messy in-depth check each time I want to know if a collection is permissioned or not.

- [ ] Revisit and document in case I have a change of heart.

Reply:
> I would suggest making them private fields and implementing a constructor that returns and error if both fields arn't provided

I dislike that in a public data struct, and it doesn't really remove the need for validation as we have to handle json-deserialized Policies (via http client) anyway.

Reply:
> I dislike that in a public data struct, and it doesn't really remove the need for validation as we have to handle json-deserialized Policies (via http client) anyway.

Good point. I hadn't thought through the fact that this need to be public for marshalling purposes.

Reply:
> todo: This really feels like a waste of the `Option` type, this whole function should be replaced with `policy.HasValue()` IMO

I revisited the original todo, and want to keep this function (might rename to `HasPolicy` though according to another suggestion).